hack/benchmark/internal/operation/insert.go

```diff
@@ -0,0 +1,127 @@
+package operation
+
+import (
+	"context"
+	"io"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/vdaas/vald/apis/grpc/v1/payload"
+	"github.com/vdaas/vald/hack/benchmark/internal/assets"
+	"github.com/vdaas/vald/internal/net/grpc/codes"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+)
+
+func (o *operation) Insert(b *testing.B, ctx context.Context, ds assets.Dataset) (insertedNum int) {
+	b.ResetTimer()
+	b.Run("Insert", func(b *testing.B) {
+		req := &payload.Insert_Request{
+			Vector: &payload.Object_Vector{},
+			Config: &payload.Insert_Config{
+				SkipStrictExistCheck: false,
+			},
+		}
+		b.ResetTimer()
+
+		for i := 0; i < b.N; i++ {
+			v, err := ds.Train(i % ds.TrainSize())
+			if err != nil {
+				b.Error(err)
+				continue
+			}
+
+			req.Vector.Id, req.Vector.Vector = strconv.Itoa(i), v.([]float32)
+
+			loc, err := o.client.Insert(ctx, req)
+			if err != nil {
+				st, _ := status.FromError(err)
+				if st.Code() != codes.AlreadyExists {
+					statusError(b, int32(st.Code()), st.Message(), st.Details()...)
+				}
+				continue
+			}
+			if loc == nil {
+				b.Error("returned loc is nil")
+			}
+			insertedNum++
+		}
+	})
+	return insertedNum
+}
+
+func (o *operation) StreamInsert(b *testing.B, ctx context.Context, ds assets.Dataset) int {
+	var insertedNum int64
+	b.ResetTimer()
+	b.Run("StreamInsert", func(b *testing.B) {
+		sc, err := o.client.StreamInsert(ctx)
+		if err != nil {
+			b.Fatal(err)
+		}
+
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+
+		req := &payload.Insert_Request{
+			Vector: &payload.Object_Vector{},
+			Config: &payload.Insert_Config{
+				SkipStrictExistCheck: false,
+			},
+		}
+		b.ResetTimer()
+
+		go func() {
+			defer wg.Done()
+
+			for {
+				res, err := sc.Recv()
+				if err == io.EOF {
+					return
+				}
+
+				if err != nil {
+					// When the StreamInsert handler on the Server side returns an error, the error will be returned to Recv method.
+					// In the case of multiple executions, such as benchmarking, an error will occur even if AlreadyExist occurs for some of them.
+					// To prevent this, we close the stream early when an error occurs.
+					return
```

Comment:
should we fail the test here?

Reply:
we should not fail in this case.
The reasons are explained below.
https://github.com/vdaas/vald/pull/1430/files#r708936100

Reply:
Even the already exists happens, is it should handle by the response, instead of recv() function?

Reply:
yes, I think so.

Reply:
then I think we  can fail the test here?

Reply:
If an error occurs in the stream, you will get the error on line 94.

Reply:
Related comment: https://github.com/vdaas/vald/pull/1415/files#r716356337

Reply:
@hlts2 I mean if the expected error (vector not found, etc) occurred, the error should be handle in line 91.
I think in line 77, the error return should be an unexpected error(other then EOF) and we should fail the test once the error occurred, so I suggest to fail the test here.
```suggestion
					b.Error(err)
```